docs/src/examples/gradient_based_optimization.md

```diff
@@ -0,0 +1,180 @@
+# Example for a Gradient Based Optimization
+
+![Optimization Iteration Plot](../assets/Gradient_Optimizer_iterations.svg)
+
+This example shows how to perform a gradient based optimization on a TASOPT model. It uses the IPOPT optimization module using a central, relative finite difference sensitivity module.
+
+**_NOTE:_**  We are currently working on a autodiff based sensitivity module.
+
+For this example, the design variables are:
+
+1. Aspect Ratio: `AR`
+2. Lift Coefficient: `Cl`  
+3. Wing Sweep: `Λ`
+4. Tt4 at Cruise: `Tt4`
+5. Fan Pressure ratio: `pif`
+
+## Initialiation and loading models
+
+Start the script importing the necessary modules.
+
+```julia
+# Import modules
+using TASOPT
+using JuMP
+using Ipopt
+using PythonPlot
+using Test
+include(__TASOPTindices__)
+```
+
+Initialize arrays used for plotting as well as the finite difference parameter
+
+```julia
+design_variables = []
+objective_array = []
+
+epsilon = 1e-5
+```
+
+Load the default `aircraft` model and size it to get initial values:
+
+```julia
+default_model = load_default_model()
+size_aircraft!(default_model)
+```
+
+## Setting Optimization Parameters
+
+### Set the input parameters
+
+```julia
+input_params = [
+    :(ac.parg[igAR]), 
+    :(ac.para[iaCL,ipcruise1:ipcruise2,1]),
+    :(ac.parg[igsweep]),
+    :(ac.pare[ieTt4, ipcruise1:ipcruise2, 1]),
+    :(ac.pare[iepif, ipcruise1, 1]) ,
+]
+
+# Formatting params for usage 
+params = map(p -> TASOPT.format_params(TASOPT.expr_to_string(p)), input_params)
+```
+
+### Set the Upper and Lower limits for all design variables as well as initial values
+
+```julia
+lower      = [9.0 , 0.53, 25.0, 1400.0, 1.25]
+upper      = [11.0, 0.60, 30.0, 1650.0, 2.0 ] 
+initial    = [10.5, 0.57, 26.0, 1580.0, 1.685]
+
+```
+
+## Objective Function
+
+```julia
+function size_ac(x...)
+    # println("State: ",x)
+    ac = deepcopy(default_model)
+    # Set params
+    for (i,x_i) in enumerate(x)
+        field_path, index = params[i]
+        TASOPT.setNestedProp!(ac, field_path, x_i, index)
```

Comment:
May be nice to add a comment explaining what this does. It may not be obvious that we need this convoluted syntax to access the par fields.